### PR TITLE
Te auth

### PR DESCRIPTION
**Pull request by Testell**

### CODE REVIEW

File: app/controllers/comments_controller.rb
Comment (jessjno) [inline app/controllers/comments_controller.rb:1]: nice job incorporating the redirect back and the alert

File: app/views/users/show.html.erb
Comment (jessjno) [inline app/views/users/show.html.erb:1]: nice formatting

File: app/views/photos/_photo.html.erb
Comment (jessjno) [inline app/views/photos/_photo.html.erb:1]: formatting looks clean

File: app/controllers/photos_controller.rb
Comment (jessjno) [inline app/controllers/photos_controller.rb:1]: good job added the private method

**Review by jessjno — APPROVED** [review]
looks good

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -67,4 +68,11 @@ def set_comment
     def comment_params
       params.require(:comment).permit(:author_id, :photo_id, :body)
     end
+
+    def is_an_authorized_user
+      @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+      if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@comment.owner)
```
Comment (marywebby) [inline app/controllers/comments_controller.rb:74]: Good job making the changes here!

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
+  before_action :ensure_current_user_is_owner, only: [:destroy. :update, :edit]
```
Comment (marywebby) [inline app/controllers/photos_controller.rb:3]: 👍

File: app/views/photos/_photo.html.erb [hunk 3]
```diff
@@ -7,12 +7,14 @@
     </h2>
 
     <div>
-      <%= link_to edit_photo_path(photo), class: "btn btn-link btn-sm text-muted" do %>
-        <i class="fas fa-edit fa-fw"></i>
-      <% end %>
+      <% if current_user == photo.owner %>
```
Comment (marywebby) [inline app/views/photos/_photo.html.erb:10]: Good job with the linking back!

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -4,16 +4,18 @@
...
 
-<div class="row mb-2">
-  <div class="col-md-6 offset-md-3">
-    <%= render "users/profile_nav", user: @user %>
-  </div>
-</div>
-
-<% @user.own_photos.each do |photo| %>
-  <div class="row mb-4">
+<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user)  %>
```
Comment (marywebby) [inline app/views/users/show.html.erb:7]: NIce job adding in current_user here!

**Review by marywebby — CHANGES_REQUESTED** [review]
It looks like you're missing the changes needed in the routes.rb file, please make those changes and then re-request me!

**Review by marywebby — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**